model/riscv_insts_vext_mem.sail

```diff
@@ -0,0 +1,1536 @@
+  VLE8      <-> "8",
+  VLE16     <-> "16",
+  VLE32     <-> "32",
+  VLE64     <-> "64"
+}
+
+mapping encdec_vlewidth : vlewidth <-> bits(3) = {
+  VLE8      <-> 0b000,
+  VLE16     <-> 0b101,
+  VLE32     <-> 0b110,
+  VLE64     <-> 0b111
+}
+
+mapping vlewidth_bytesnumber : vlewidth <-> {|1, 2, 4, 8|} = {
+  VLE8      <-> 1,
+  VLE16     <-> 2,
+  VLE32     <-> 4,
+  VLE64     <-> 8
+}
+
+mapping vlewidth_pow : vlewidth <-> {|3, 4, 5, 6|} = {
+  VLE8      <-> 3,
+  VLE16     <-> 4,
+  VLE32     <-> 5,
+  VLE64     <-> 6
+}
+
+mapping bytes_wordwidth : {|1, 2, 4, 8|} <-> word_width = {
+  1 <-> BYTE,
+  2 <-> HALF,
+  4 <-> WORD,
+  8 <-> DOUBLE
+}
+
+/* ******************** Vector Load Unit-Stride Normal (nf=0, mop=0, lumop=0) ******************** */
+union clause ast = VLETYPE : (bits(1), regidx, vlewidth, regidx)
+
+mapping clause encdec = VLETYPE(vm, rs1, width, vd) if haveRVV()
+  <-> 0b000 @ 0b0 @ 0b00 @ vm @ 0b00000 @ rs1 @ encdec_vlewidth(width) @ vd @ 0b0000111 if haveRVV()
+
+val process_vle : forall 'b 'n 'p, ('b in {1, 2, 4, 8}) & ('n >= 0). (bits(1), regidx, int('b), regidx, int('p), int('n)) -> Retired effect {escape, rmem, rmemt, rreg, undef, wmv, wmvt, wreg}
+function process_vle (vm, vd, load_width_bytes, rs1, EMUL_pow, num_elem) = {
+  let width_type : word_width = bytes_wordwidth(load_width_bytes);
+  let vm_val : vector('n, dec, bool) = read_vmask(num_elem, vm, 0b00000);
+  let vd_val : vector('n, dec, bits('b * 8)) = read_vreg(num_elem, load_width_bytes * 8, EMUL_pow, vd);
+  total : vector('n, dec, bits('b * 8)) = undefined;
+  mask  : vector('n, dec, bool) = undefined;
+  status : Retired = RETIRE_SUCCESS;
+
+  (total, mask) = init_masked_result(num_elem, load_width_bytes * 8, EMUL_pow, vd_val, vm_val);
+
+  foreach (i from 0 to (num_elem - 1)) {
+    if status != RETIRE_FAIL then {
+      if mask[i] then {
+        vstart = to_bits(16, i);
+        let elem_offset = i * load_width_bytes;
+        match ext_data_get_addr(rs1, to_bits(sizeof(xlen), elem_offset), Read(Data), width_type) {
+          Ext_DataAddr_Error(e)  => { ext_handle_data_check_error(e); status = RETIRE_FAIL },
+          Ext_DataAddr_OK(vaddr) => 
+            if check_misaligned(vaddr, width_type)
+            then { handle_mem_exception(vaddr, E_Load_Addr_Align()); status = RETIRE_FAIL }
+            else match translateAddr(vaddr, Read(Data)) {
+              TR_Failure(e, _)     => { handle_mem_exception(vaddr, e); status = RETIRE_FAIL },
+              TR_Address(paddr, _) => {
+                match mem_read(Read(Data), paddr, load_width_bytes, false, false, false) {
+                  MemValue(result) => total[i] = result,
+                  MemException(e)  => { handle_mem_exception(vaddr, e); status = RETIRE_FAIL }
+                }
+              }
+            }
+        }
+      }
+    }
+  };
+
+  if status == RETIRE_SUCCESS then write_vreg(num_elem, load_width_bytes * 8, EMUL_pow, vd, total);
+  vstart = EXTZ(0b0);
+  status
```

Comment:
I think there are issues in this place and other vector memory instructions.
1. `vstart` should be set back to 0 only if there is no fault. If this condition is added, then vstart will be the element that generates the fault after execution, or 0 when there is no fault.
2. Elements should be written one by one in the iteration instead of written together (to make sure the elements before the fault are written into vd).
I'll modify this part.

For the verification, we currently do not have fault tests on vector load/stores, so this part has not been tested before.

Reply:
This part has been modified.